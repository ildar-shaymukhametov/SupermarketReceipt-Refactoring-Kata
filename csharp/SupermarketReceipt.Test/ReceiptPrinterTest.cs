using ApprovalTests;
using ApprovalTests.Reporters;
using Xunit;

namespace SupermarketReceipt.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class ReceiptPrinterTest
    {
        readonly Product _toothbrush = new Product("toothbrush", ProductUnit.Each);
        readonly Product _apples = new Product("apples", ProductUnit.Kilo);
        Receipt _receipt = new Receipt();

        [Fact]
        public void oneLineItem()
        {
            _receipt.AddProduct(_toothbrush, 1, 0.99, 0.99);
            Approvals.Verify(new ReceiptPrinter().PrintReceipt(_receipt));
        }
        
        [Fact]
        public void quantityTwo()
        {
            _receipt.AddProduct(_toothbrush, 2, 0.99, 0.99 * 2);
            Approvals.Verify(new ReceiptPrinter().PrintReceipt(_receipt));
        }
        
        [Fact]
        public void looseWeight()
        {
            _receipt.AddProduct(_apples, 2.3, 1.99, 1.99 * 2.3);
            Approvals.Verify(new ReceiptPrinter().PrintReceipt(_receipt));
        }

        [Fact]
        public void total()
        {

            _receipt.AddProduct(_toothbrush, 1, 0.99, 2 * 0.99);
            _receipt.AddProduct(_apples, 0.75, 1.99, 1.99 * 0.75);
            Approvals.Verify(new ReceiptPrinter().PrintReceipt(_receipt));
        }

        [Fact]
        public void discounts()
        {
            _receipt.AddDiscount(new Discount(_apples, "3 for 2", 0.99));
            Approvals.Verify(new ReceiptPrinter().PrintReceipt(_receipt));
        }

        [Fact]
        public void printWholeReceipt()
        {
            _receipt.AddProduct(_toothbrush, 1, 0.99, 0.99);
            _receipt.AddProduct(_toothbrush, 2, 0.99, 2 * 0.99);
            _receipt.AddProduct(_apples, 0.75, 1.99, 1.99 * 0.75);
            _receipt.AddDiscount(new Discount(_toothbrush, "3 for 2", 0.99));
            Approvals.Verify(new ReceiptPrinter().PrintReceipt(_receipt));
        }
        
        [Fact]
        public void printWholeReceipt_html()
        {
            _receipt.AddProduct(_toothbrush, 1, 0.99, 0.99);
            _receipt.AddProduct(_toothbrush, 2, 0.99, 2 * 0.99);
            _receipt.AddProduct(_apples, 0.75, 1.99, 1.99 * 0.75);
            _receipt.AddDiscount(new Discount(_toothbrush, "3 for 2", 0.99));
            Approvals.Verify(new HtmlReceiptDecorator(new ReceiptPrinter()).PrintReceipt(_receipt));
        }
    }
}