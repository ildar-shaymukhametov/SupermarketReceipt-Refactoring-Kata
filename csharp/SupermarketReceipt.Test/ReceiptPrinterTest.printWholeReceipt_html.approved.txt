<table>
<tr><td>toothbrush                          0.99</td></tr>
<tr><td>toothbrush                          1.98</td></tr>
<tr><td>  0.99 * 2</td></tr>
<tr><td>apples                              1.49</td></tr>
<tr><td>  1.99 * 0.750</td></tr>
<tr><td>3 for 2(toothbrush)                 0.99</td></tr>
<tr><td>Total:                              5.45</td></tr>
</table>
